TwitchDownloaderCore/ChatDownloader.cs

```diff
@@ -244,13 +244,36 @@ private static List<Comment> ConvertComments(CommentVideo video, ChatFormat form
             return returnList;
         }
 
-        public async Task DownloadAsync(CancellationToken cancellationToken)
+        public async Task DownloadAsync(CancellationToken cancellationToken, bool allowOverwrite = true)
         {
             if (string.IsNullOrWhiteSpace(downloadOptions.Id))
             {
                 throw new NullReferenceException("Null or empty video/clip ID");
             }
 
+            if(allowOverwrite) 
+            {
+                while (File.Exists(downloadOptions.Filename))
+                {
+                    _progress.SetStatusWarning("File already exists. Do you want to overwrite it? (Confirm/Reject)");
+                    var response = Console.ReadLine()?.Trim().ToLower();
+        
+                    if (response == "confirm")
+                    {
+                        break;
+                    }
+                    else if (response == "reject")
+                    {
+                        _progress.SetStatusWarning("Operation aborted. File not overwritten.");
+                        return;
+                    }
+                    else
+                    {
+                        _progress.SetStatusWarning("Invalid response. Please enter 'confirm' to overwrite or 'reject' to cancel.");
+                    }
+                }
+            }
+
```

Comment:
DRY - Don't repeat yourself. You have the exact same block of code in 5 different files.